### PR TITLE
PV2-54: required payments domain impl

### PR DESCRIPTION
**Pull request by kuzinator**
Required payments can handle payments due messages, the only existing acceptance test passing with a little crutch of converting and earning to payments due event. The service will need to be looked at again when acceptance tests converted. PaymentHistoryRepository assumes schema exists.

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.Domain.UnitTests/Services/Act2PaymentDueProcessorTest.cs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+using SFA.DAS.Payments.Model.Core;
+using SFA.DAS.Payments.Model.Core.OnProgramme;
+using SFA.DAS.Payments.PaymentsDue.Messages.Events;
+using SFA.DAS.Payments.RequiredPayments.Domain.Entities;
+using SFA.DAS.Payments.RequiredPayments.Domain.Services;
+
+namespace SFA.DAS.Payments.RequiredPayments.Domain.UnitTests.Services
+{
+    [TestFixture]
+    public class Act2PaymentDueProcessorTest
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.Domain.UnitTests/Services/Act2PaymentDueProcessorTest.cs]: Rename to ApprenticeshipContractTyp2PaymentDueProcessorTest

File: src/SFA.DAS.Payments.RequiredPayments.Domain/Interfaces/IAct2PaymentDueProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using SFA.DAS.Payments.PaymentsDue.Messages.Events;
+using SFA.DAS.Payments.RequiredPayments.Domain.Entities;
+using SFA.DAS.Payments.RequiredPayments.Messages.Events;
+
+namespace SFA.DAS.Payments.RequiredPayments.Domain
+{
+    public interface IAct2PaymentDueProcessor
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.Domain/Interfaces/IAct2PaymentDueProcessor.cs]: Rename to IApprenticeshipContractTyp2PaymentDueProcessor

File: src/SFA.DAS.Payments.RequiredPayments.Domain/Interfaces/IAct2PaymentDueProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    {
+        ApprenticeshipContractType2RequiredPaymentEvent ProcessPaymentDue(ApprenticeshipContractType2PaymentDueEvent paymentDue, Payment[] paymentHistory);
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.Domain/Interfaces/IAct2PaymentDueProcessor.cs]: There is no need for the domain to use the incoming message.  The domain only needs the amount due and the period.

File: src/SFA.DAS.Payments.RequiredPayments.Domain/Services/Act2PaymentDueProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Linq;
+using SFA.DAS.Payments.PaymentsDue.Messages.Events;
+using SFA.DAS.Payments.RequiredPayments.Domain.Entities;
+using SFA.DAS.Payments.RequiredPayments.Messages.Events;
+
+namespace SFA.DAS.Payments.RequiredPayments.Domain.Services
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.Domain/Services/Act2PaymentDueProcessor.cs]: Rename to ApprenticeshipContractTyp2PaymentDueProcessor

File: src/SFA.DAS.Payments.RequiredPayments.Messages/Events/IRequiredPayment.cs [hunk 5]
```diff
@@ -1,9 +1,13 @@
 ﻿using SFA.DAS.Payments.Messages.Core.Events;
+using SFA.DAS.Payments.Model.Core;
 
 namespace SFA.DAS.Payments.RequiredPayments.Messages.Events
 {
     public interface IRequiredPayment : IPaymentsEvent
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.Messages/Events/IRequiredPayment.cs]: This should be derived from IPeriodisedEarningEvent.

File: src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/Act2PaymentDueEventHandler.cs [hunk 6]
```diff
@@ -3,27 +3,27 @@
...
 using System.Threading.Tasks;
 using SFA.DAS.Payments.Application.Infrastructure.Logging;
+using SFA.DAS.Payments.PaymentsDue.Messages.Events;
+using SFA.DAS.Payments.RequiredPayments.Domain;
 
 
 namespace SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService.Handlers
 {
-    public class PayableEarningEventHandler : IHandleMessages<ApprenticeshipContractType2EarningEvent>
+    public class Act2PaymentDueEventHandler : IHandleMessages<ApprenticeshipContractType2PaymentDueEvent>
```
Comment (abroodie) [inline src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/Act2PaymentDueEventHandler.cs]: rename to ApprenticeshipContractType2PaymentDueEventHandler

**Comment by kuzinator** [comment]
Updated class names and domain service as per requested changes.

**Pull request merged**